e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -360,15 +399,19 @@ if (-not $spExists) {
 # Adding retries so that we can grant the permissions successfully without re-running the script.
 $retry = $true
 $retryCount = 1;
-while ($retry -eq $true) {
-    try {
+while ($retry)
+{
+    try
+    {
         Write-Host "`nGranting $appId Reader role assignment to the $Resourcegroup resource group - Trial $retryCount"
         az role assignment create --role Reader --assignee $appId --resource-group $ResourceGroup --output none
         $retry = $false
     }
-    catch {
+    catch
+    {
         $retryCount++
-        if ($retryCount -eq 10) {
+        if ($retryCount -ge 10)
```

Comment:
Using greater than or equal to in order to avoid any possible case where we didn't catch it on exactly 10. Obviously, we don't expect this code to do that, but it is a general best practice when testing boundaries.